lib/cart/cart_page.dart

```diff
@@ -7,11 +12,45 @@ class CartPage extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    return BlocProvider<CartCubit>(
+      create: (context) => CartCubit()..loadCart(),
+      child: const CartView(),
+    );
+  }
+}
+
+class CartView extends StatelessWidget {
+  const CartView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final itemCount =
+        context.select((CartCubit cubit) => cubit.state.cart?.totalItems);
+
     return Scaffold(
       appBar: AppBar(
-        title: const Text('Cart'),
+        title: Text('Cart ${itemCount != null ? '($itemCount items)' : ''}'),
+      ),
+      body: BlocBuilder<CartCubit, CartState>(
+        builder: (context, state) {
+          switch (state.status) {
+            case Status.pending:
+              return const Center(
+                child: CircularProgressIndicator(),
+              );
+            case Status.success:
+              return const CartContent();
+            case Status.failure:
+              return const Center(
+                child: Text('Failed to load cart'),
+              );
+            default:
```

Comment:
I would remove this `default` clause. 

Without it, Dart generates a compile error if the developper adds a new status, it is a nice thing.